itou/approvals/test_notify_pole_emploi.py

```diff
@@ -185,3 +190,38 @@ def test_notification_goes_to_pending_if_job_application_is_not_accepted(self):
         approval.notify_pole_emploi(at=now)
         approval.refresh_from_db()
         self.assertEqual(approval.pe_notification_status, "notification_pending")
+
+
+class ApprovalsSendToPeManagementTestCase(ApprovalNotifyPoleEmploiBaseTest):
+    @patch.object(Approval, "notify_pole_emploi")
+    @patch("itou.approvals.management.commands.send_approvals_to_pe.sleep")
+    def test_invalid_job_seeker_for_pole_emploi(self, sleep_mock, notify_mock):
+        stdout = io.StringIO()
+        ApprovalFactory(with_jobapplication=False)
+        retry_approval = ApprovalFactory(
+            start_at=datetime.datetime.today().date() - datetime.timedelta(days=1),
+            with_jobapplication=True,
+            pe_notification_status="notification_should_retry",
+        )
+        pending_approval = ApprovalFactory(with_jobapplication=True)
+        management.call_command(
+            "send_approvals_to_pe",
+            wet_run=True,
+            delay=3,
+            stdout=stdout,
+        )
+        stdout.seek(0)
+        output = stdout.readlines()
+        self.assertEqual(
+            output,
+            [
+                "approvals needing to be sent count=2, batch count=100\n",
+                f"approvals={pending_approval} start_at={pending_approval.start_at.isoformat()} "
+                "pe_state=notification_pending\n",
+                f"approvals={retry_approval} start_at={retry_approval.start_at.isoformat()} "
+                "pe_state=notification_should_retry\n",
+            ],
+        )
```

Comment:
C'est pas tout à fait équivalent mais tu as `io.String.getvalue()` qui te renvois tout le buffer, ça évite d'avoir à _seek_ (et donc modifier la position) puis _read_.
```suggestion
        self.assertEqual(
            stdout.getvalue().split("\n"),
            [
                "approvals needing to be sent count=2, batch count=100",
                f"approvals={pending_approval} start_at={pending_approval.start_at.isoformat()} pe_state=notification_pending",
                f"approvals={retry_approval} start_at={retry_approval.start_at.isoformat()} pe_state=notification_should_retry",
            ],
        )
```